evm/vm/transaction_context.py

```diff
@@ -15,12 +19,13 @@ class BaseTransactionContext:
     _origin = None
     _log_counter = None
 
-    def __init__(self, gas_price, origin):
+    def __init__(self, gas_price, origin, data):
         validate_uint256(gas_price, title="TransactionContext.gas_price")
         self._gas_price = gas_price
         validate_canonical_address(origin, title="TransactionContext.origin")
         self._origin = origin
         self._log_counter = itertools.count()
+        self.data = data
```

Comment:
Seems odd to me to add this data here. I guess I don't totally get what TransactionContext is, because it seems like it's just adding more and more things until it eventually just becomes a copy of Transaction. Why not use the original Transaction directly?

Reply:
@carver
The `data` field was required to make the intrinsic gas calculation.  Adding the intrinsic gas to the transaction context does create an awkward api, as you pointed out.

The move of the intrinsic gas to the transaction context began with @pipermerriam suggestion after I duplicated some BaseTransaction attributes in BaseUnsignedTransaction.  Maybe it would make sense to have a shared abstract class between BaseUnsignedTransaction and BaseTransaction containing the required attributes to make the gas estimation with unsigned transactions.  Or come up with a clever way to add the intrinsic gas to SpoofTransaction.

Reply:
I think I'm going to flip-flop my opinion on this.

The purpose of `TransactionContext` is to provide the transaction level data necessary for vm execution, i.e. the `gas_price` and the `origin` address.

@dylanjw I think a common base for the transaction classes which implements the intrinsic gas calculations should be fine.  I don't think there is anything in the signed vs unsigned that restricts either from being able to compute the intrinsic gas costs.